src/lib/transaction_logic/mina_transaction_logic.ml

```diff
@@ -248,6 +248,14 @@ module Transaction_applied = struct
         c.coinbase.status
 end
 
+(* ZEKO NOTE: issue #64 *)
+type env = < mark_shifted_and_get_previous_shiftedness : Account_id.t -> bool >
```

Comment:
why is this an object type?

Reply:
ah so that it has encapsulated state of shifts?

Reply:
Immediate objects are anonymous records. I define env type many types, and this means they are all equal.
If I just used a record, they would be incompatible, and it would mean you would need to specify which you mean when implementing the env. 

Another solution is to define a normal record in a single place.